examples/azure-provider.md

```diff
@@ -0,0 +1,58 @@
+# Azure provider
+
+## Overview
+
+The Azure Provider in Secret-init can load secrets from Azure Key Vault. This provider interfaces with Azure's API, to fetch and load secrets.
+
+## Prerequisites
+
+- Golang `>= 1.21`
+- Makefile
+- Access to Azure services
+
+## Environment setup
+
+```bash
+# Secret-init requires atleast this environment variable to be set properly
+export AZURE_KEY_VAULT_URL
+```
+
+The options provided in the [Azure SDK Authentication Guide](https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication?tabs=bash#2-authenticate-with-azure) are all viable when using `secret-init`.
+This includes:
+
+- Authenticating with Azure using environment variables.
+- A service principal with a client secret.
+- A service principal with a certificate.
+- A managed identities for Azure resources, and interactive browser authentication.
```

Comment:
maybe we should remove interactive browser auth? not sure if this will actually work in our use-cases